src/Core/Datatypes/Dyadic3DTensor.h

```diff
@@ -0,0 +1,184 @@
+        (*this)(1, 2) = (*this)(2, 1) = v5;
+        parent::buildEigens();
+      }
+
+      Number linearCertainty()
+      {
+        auto eigvals = parent::getEigenvalues();
+        return (eigvals[0] - eigvals[1]) / parent::eigenValueSum();
+      }
+
+      Number planarCertainty()
+      {
+        auto eigvals = parent::getEigenvalues();
+        return 2.0 * (eigvals[1] - eigvals[2]) / parent::eigenValueSum();
+      }
+
+      Number sphericalCertainty()
+      {
+        auto eigvals = parent::getEigenvalues();
+        return 3.0 * eigvals[2] / parent::eigenValueSum();
+      }
+
+      DenseColumnMatrixGeneric<Number> mandel()
+      {
+        auto eigvals = parent::getEigenvalues();
+        auto eigvecs = parent::getEigenvectors();
+
+        for (size_t i = 0; i < DIM_; ++i)
+          eigvecs[i] *= eigvals[i];
+
+        const double sqrt2 = std::sqrt(2);
+        DenseColumnMatrixGeneric<Number> mandel({eigvecs[0][0], eigvecs[1][1], eigvecs[2][2],
+            eigvecs[0][1] * sqrt2, eigvecs[0][2] * sqrt2, eigvecs[1][2] * sqrt2});
+        return mandel;
+      }
+
+     private:
+      const size_t DIM_ = 3;
+
+      std::vector<DenseColumnMatrixGeneric<Number>> convertNativeVectorsToEigen(
+          const std::vector<Core::Geometry::Vector>& vecs)
+      {
+        std::vector<DenseColumnMatrixGeneric<Number>> outVecs(vecs.size());
+        for (size_t i = 0; i < vecs.size(); ++i)
+        {
+          outVecs[i] = DenseColumnMatrixGeneric<Number>(DIM_);
+          for (size_t j = 0; j < DIM_; ++j)
+            outVecs[i][j] = vecs[i][j];
+        }
+        return outVecs;
+      }
+    };
+
+    template <typename Indexable>
+    Dyadic3DTensor symmetricTensorFromSixElementArray(const Indexable& array)
+    {
+      if (array.size() != 6) THROW_INVALID_ARGUMENT("This function requires 6 values.");
+      return Dyadic3DTensor(array[0], array[1], array[2], array[3], array[4], array[5]);
+    }
+
+    template <typename Number>
+    Dyadic3DTensor symmetricTensorFromSixElementArray(const std::initializer_list<Number>& array)
+    {
+      std::vector<Number> vec = array;
+      if (vec.size() != 6) THROW_INVALID_ARGUMENT("This function requires 6 values.");
+      return Dyadic3DTensorGeneric<double>(vec[0], vec[1], vec[2], vec[3], vec[4], vec[5]);
+    }
+
+    template <typename Indexable>
+    Dyadic3DTensor symmetricTensorFromNineElementArray(const Indexable& array)
+    {
+      if (array.size() != 9) THROW_INVALID_ARGUMENT("This function requires 9 values.");
+      return Dyadic3DTensor(array[0], array[1], array[2], array[4], array[5], array[8]);
+    }
+
+    template <typename Number>
+    Dyadic3DTensor symmetricTensorFromNineElementArray(const std::initializer_list<Number>& array)
+    {
+      std::vector<Number> vec = array;
+      if (vec.size() != 9) THROW_INVALID_ARGUMENT("This function requires 9 values.");
+      return Dyadic3DTensorGeneric<double>(vec[0], vec[1], vec[2], vec[4], vec[5], vec[8]);
+    }
```

Comment:
These convenience functions can be made even more convenient with a variadic function, which would move the runtime size check to compile time. It would look something like:
```
template <typename ...Number>
Dyadic3DTensor buildSymmetricTensorFrom(Number ... n)
{
   return Dyadic3DTensor(n...);
}
```
There would need to be some more machinery to get all the cases working (like the 9->6 element case), but it would be much easier to use for algo writers.